acceptance-tests/src/test/java/tech/pegasys/eth2signer/tests/filecoin/FcSignMessageAcceptanceTest.java

```diff
@@ -0,0 +1,96 @@
+import tech.pegasys.eth2signer.tests.signing.SigningAcceptanceTestBase;
+import tech.pegasys.teku.bls.BLSKeyPair;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.bls.BLSSecretKey;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ValueNode;
+import com.github.arteam.simplejsonrpc.core.domain.Request;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import org.apache.tuweni.bytes.Bytes;
+import org.junit.jupiter.api.Test;
+
+public class FcSignMessageAcceptanceTest extends SigningAcceptanceTestBase {
+
+  private static final String PRIVATE_KEY = "z38sVnSEnswoEHFC9e4g/aPk96c1NvXt425UKv/tKz0=";
+
+  private static final MetadataFileHelpers metadataFileHelpers = new MetadataFileHelpers();
+  private static final BLSSecretKey key =
+      BLSSecretKey.fromBytes(Bytes.fromBase64String(PRIVATE_KEY));
+  private static final BLSKeyPair keyPair = new BLSKeyPair(key);
+  private static final BLSPublicKey publicKey = keyPair.getPublicKey();
+  private static final FilecoinNetwork network = FilecoinNetwork.TESTNET;
+  final FilecoinAddress sender = FilecoinAddress.blsAddress(publicKey.toBytes());
+
+  @Test
+  void fcSignMessageReturnsASignedMessage() {
+    final String configFilename = publicKey.toString().substring(2);
+    final Path keyConfigFile = testDirectory.resolve(configFilename + ".yaml");
+    metadataFileHelpers.createUnencryptedYamlFileAt(
+        keyConfigFile, key.getSecretKey().toBytes().toUnprefixedHexString(), KeyType.BLS);
+
+    setupSigner();
+
+    final ValueNode id = JsonNodeFactory.instance.numberNode(1);
+    ObjectMapper mapper = new ObjectMapper();
+
+    final Map<String, Object> paramMap = new HashMap<>();
+
+    paramMap.put("Version", 9);
+    paramMap.put("To", "t01234");
+    paramMap.put("From", sender.encode(network));
+    paramMap.put("Nonce", 42);
+    paramMap.put("Value", "0");
+    paramMap.put("GasPrice", "0");
+    paramMap.put("GasLimit", 9);
+    paramMap.put("Method", 1);
+    paramMap.put("Params", "Ynl0ZSBhcnJheQ==");
+
+    final Request request =
+        new Request(
+            "2.0", "Filecoin.WalletSignMessage", mapper.convertValue(paramMap, JsonNode.class), id);
+    final Response response =
+        given().baseUri(signer.getUrl()).body(request).post(JSON_RPC_PATH + "/filecoin");
+
+    response
+        .then()
+        .statusCode(200)
+        .contentType(ContentType.JSON)
+        .body("jsonrpc", equalTo("2.0"), "id", equalTo(id.asInt()));
+
+    final FilecoinSignedMessage signedMessage =
+        response.body().jsonPath().getObject("result", FilecoinSignedMessage.class);
+
+    assertThat(signedMessage.getSignature().getType()).isEqualTo(2);
```

Comment:
even if canned values are used, would be good to verify that signature data is what we expect and that we also return the original message.

Reply:
done